agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -4563,6 +4562,41 @@ bool backhaul_manager::handle_backhaul_steering_request(ieee1905_1::CmduMessageR
     return send_cmdu_to_bus(cmdu_tx, controller_bridge_mac, bridge_info.mac);
 }
 
+bool backhaul_manager::create_backhaul_steering_response(
+    const wfa_map::tlvBackhaulSteeringResponse::eResultCode &result_code)
+{
+    auto cmdu_tx_header =
+        cmdu_tx.create(0, ieee1905_1::eMessageType::BACKHAUL_STEERING_RESPONSE_MESSAGE);
+    if (!cmdu_tx_header) {
+        LOG(ERROR) << "Failed to create Backhoul STA Steering Response message";
+        return false;
+    }
+
+    auto bh_steering_resp_tlv = cmdu_tx.addClass<wfa_map::tlvBackhaulSteeringResponse>();
+    if (!bh_steering_resp_tlv) {
+        LOG(ERROR) << "Couldn't addClass<wfa_map::tlvBackhaulSteeringResponse>";
+        return false;
+    }
+
+    auto active_hal = get_wireless_hal();
+    if (!active_hal) {
+        LOG(ERROR) << "Couldn't get active HAL";
+        return false;
+    }
+
+    sMacAddr sta_mac;
+    auto interface = active_hal->get_iface_name();
+    get_iface_mac(interface, sta_mac);
+
+    LOG(DEBUG) << "Interface: " << interface << "MAC: " << tlvf::mac_to_string(sta_mac);
+
+    bh_steering_resp_tlv->target_bssid()         = tlvf::mac_from_string(active_hal->get_bssid());
+    bh_steering_resp_tlv->backhaul_station_mac() = sta_mac;
+    bh_steering_resp_tlv->result_code()          = result_code;
```

Comment:
If result code is failure, we also have to provide an error code TLV with the error code. See the 4th paragraph of section 12.

So I think it's better to pass the error code directly, and to set the result code (and include the error TLV) depending on whether the error code is 0 or not.

Reply:
You are right. Added creation for ErrorCodeTlv